engine/FindLinuxDistro.cmake

```diff
@@ -0,0 +1,51 @@
+# - FindLinuxDistro.cmake
+# This module detects the Linux Distro and package build tooling, and provides
+# some variables for the primary CMake script to utilize to decide what to build.
+#
+# ========================
+#  Example Usage:
+#
+#	FIND_PACKAGE(LinuxDistro, REQUIRED)
+#
+# Output Variables:
+#	LSB_EXISTS
+#		Is the distro an LSB Compliant Distro?
+#
+#	LSB_LINUX_DISTRIBUTION
+#		Using LSB Tooling, what is the host distro family?
+#
+#	LSB_LINUX_DISTRIBUTION_CODENAME
+#		Using LSB Tooling, what is the host distro code name?
+#
+#	VS_CAN_BUILD_DEB
+#		Does the host have tooling installed to build Debian Packages?
+#
+
+# lsb_release -i --> but then it'll find `Ubuntu` instead of Debian so will have to map more
+
+# we can probably more reliably detect via checking for different packager commands
+# `lsb_release` doesn't always work correctly, so it's easier to read the underlying file
+# if it exists. It doesn't always exist. Debian doesn't use it, but Ubuntu does.
+SET(LSB_EXISTS FALSE)
+SET(LSB_LINUX_DISTRIBUTION "Unknown")
+SET(LSB_LINUX_DISTRIBUTION_CODENAME "Unknown")
+
+# Lookup the LSB Data - this is known for Ubuntu, but also used by other distros too.
+# and may help with distro-sub-selection
+FIND_PROGRAM(LSB_RELEASE_EXEC lsb_release)
```

Comment:
What about also including the functionality from lines 180-191 of the main CMakeLists.txt file, that checks for and reads the `os-release` file, if present?

Reply:
I didn't see that. Thanks for pointing it out.